src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Prerendering.cs

```diff
@@ -33,19 +34,39 @@ protected override IComponent ResolveComponentForRenderMode([DynamicallyAccessed
         }
     }
 
+    protected override IComponentRenderMode? GetComponentRenderMode(IComponent component)
+    {
+        var componentState = GetComponentState(component);
+        var ssrRenderBoundary = GetClosestRenderModeBoundary(componentState);
+
+        if (ssrRenderBoundary is null)
+        {
+            return null;
+        }
```

Comment:
This threw an exception; I am not sure that's the right behavior. I've switched it to return null, however it feels that we are missing a way to represent a `StaticRenderMode`